src/grid_map_geo.cpp

```diff
@@ -104,41 +139,82 @@ bool GridMapGeo::initializeFromGeotiff(const std::string &path) {
 
   const OGRSpatialReference *spatial_ref = dataset->GetSpatialRef();
   coordinate_name_ = spatial_ref->GetAttrValue("geogcs");
+
   // Get image metadata
-  unsigned width = dataset->GetRasterXSize();
-  unsigned height = dataset->GetRasterYSize();
+  const auto raster_width = dataset->GetRasterXSize();
+  const auto raster_height = dataset->GetRasterYSize();
+
+  if (raster_width > max_raster_x_size_ || raster_height > max_raster_y_size_) {
+    std::cout << "Raster too big. Using a submap of size " << max_raster_x_size_ << "x" << max_raster_y_size_
+              << std::endl;
+  }
   double resolution = pixelSizeX;
-  std::cout << "Width: " << width << " Height: " << height << " Resolution: " << resolution << std::endl;
+  std::cout << "RasterX: " << raster_width << " RasterY: " << raster_height << " pixelSizeX: " << pixelSizeX
+            << std::endl;
+
+  // Limit grid map size to not exceed memory limitations
+  const auto grid_width = std::min(raster_width, max_raster_x_size_);
+  const auto grid_height = std::min(raster_height, max_raster_y_size_);
 
   // pixelSizeY is negative because the origin of the image is the north-east corner and positive
   // Y pixel coordinates go towards the south
-  const double lengthX = resolution * width;
-  const double lengthY = resolution * height;
+  const double lengthX = resolution * grid_width;
+  const double lengthY = resolution * grid_height;
   grid_map::Length length(lengthX, lengthY);
+  std::cout << "GMLX: " << lengthX << " GMLY: " << lengthY << std::endl;
+
+  //! @todo use WGS-84 as map origin if specified
+  //! @todo
+  //! @todo check the origin point is non-void (not in the middle of the ocean)
+  Corners corners;
+  if (!getGeoCorners(dataset, corners)) {
+    std::cerr << "Failed to get geographic corners of dataset!" << std::endl;
+    return false;
+  }
 
-  double mapcenter_e = originX + pixelSizeX * width * 0.5;
-  double mapcenter_n = originY + pixelSizeY * height * 0.5;
-  maporigin_.espg = ESPG::CH1903_LV03;
-  maporigin_.position = Eigen::Vector3d(mapcenter_e, mapcenter_n, 0.0);
+  if (std::isnan(maporigin_.position.x()) || std::isnan(maporigin_.position.y())) {
```

Comment:
This would just ignore the actual origin of the map, right? Wouldn't it be better to take whatever is specified in the map?

Reply:
In the SRTM1 data file, the map origin is latitude=0, longitude=0, and it's in the middle of the ocean and there is no data available. So yes, if the user does not specify a reasonable origin over land, then it will try loading tiles over the ocean which are entirely void data.

For now, I was just hoping you can launch in a non-hard-coded location (it's in the ROS params). In a future PR, I was planning to add more intelligent handling of where to launch the origin.

Reply:
How does it actually work when we are trying query specific regions of the VRT? Shouldn't we get the origin of the parsed datum when the smaller terrain file is downloaded? The way tif loading works is that you never need to specify where the origin is.

Reply:
The advantage of using a large VRT is the actual raster tile sizes and origins become data details; it's up to the user to use application-specifics heuristics for setting the origin. For most UAV applications, basing on the current UAV location should work ok.